base/client.jl

```diff
@@ -546,13 +550,23 @@ function _start()
     append!(ARGS, Core.ARGS)
     # clear any postoutput hooks that were saved in the sysimage
     empty!(Base.postoutput_hooks)
+    local ret
     try
-        exec_options(JLOptions())
+        if isdefined(Core, :Main) && isdefined(Core.Main, :main)
+            ret = Core.Main.main(ARGS)
+        elseif isassigned(REPL_MODULE_REF)
+            ret = REPL_MODULE_REF[].main(ARGS)
+        else
+            error("No entry point defined and REPL not loaded.")
```

Comment:
This seems like an API break, since we expect Julia to work correctly to run scripts via `exec_options` particularly when the REPL is not loaded.

Reply:
I'm fine with falling back to exec_options for the time being, though we may want to re-arrange later once we figure out what the compiler interface looks like.